xgboost_ray/main.py

```diff
@@ -257,16 +245,12 @@ def __init__(self,
 
         _set_omp_num_threads()
 
-    @property
-    def checkpoint_file(self) -> Optional[str]:
-        return _checkpoint_file(self.checkpoint_path, self.checkpoint_prefix,
-                                self.rank)
-
     @property
     def _save_checkpoint_callback(self):
         def callback(env):
```

Comment:
Do you know why the xgboost internal callbacks use a Class-based callback?

https://github.com/dmlc/xgboost/blob/927c316aeb0a948e9635c1ffe54d5e6ed25f6797/python-package/xgboost/callback.py#L668-L677

Reply:
This seems to be a new change, here is the legacy callback handler: https://github.com/dmlc/xgboost/blob/927c316aeb0a948e9635c1ffe54d5e6ed25f6797/python-package/xgboost/callback.py#L682

It will be added in the next release (1.3.0, current stable release 1.2.x)

It should be easy to refactor our callbacks in the future, but I'd like to keep them in the legacy format to support older installations of xgboost. We should put this on our backlog though and possibly update in (late) Q1/2021

Reply:
Cool sounds good!